### ISSUE TITLE
Some hints are missing im prompt

### ISSUE THREAD

**Issue by OlMon** [atlas-engineer/nyxt#2603]
**Describe the bug**
Some hints are missing in prompt

**Precise recipe to reproduce the issue**
There was a lot of work done with the hint and prompt logic, if this is a known bug then sorry. I tried to read through the couple last issues.

There are some hints shown on the website but are missing in the prompt. I think this is because of the new removing duplication logic. Problem is I see two hints "BH" and "BI" both are equal, I type "BH" in the prompt and then there is none or a wrong entry in the prompt.

I attach two screenshots for better understanding.
URL tested in the screenshot: https://duckduckgo.com/?q=hund&ia=web

![01-hint_available](https://user-images.githubusercontent.com/61208316/197463893-e4d4887d-74ea-42c4-9ff2-d410884986b6.png)
![02-hint_not_available](https://user-images.githubusercontent.com/61208316/197463894-84f0d159-1ca6-443f-92a0-cebd26bb2346.png)

**Information**
- OS name+version: Arch Linux
- Graphics card and driver: Nvidia
- Desktop environment / Window manager name+version: AwesomeWM Git Version
- How you installed Nyxt (Guix pack, package manager, build from source): Build from source
- Information from `show-system-information`:
```
Nyxt version: 3-pre-release-2-191-gd6bc6d9dc
Renderer version: GI-GTK
Operating system kernel: Linux 6.0.2-arch1-1
Lisp implementation: SBCL 2.2.8 (Dynamic space size: 3221225472)
Features: (:NYXT-GI-GTK :NYXT-GTK :NYXT-GI-GTK :NYXT-GTK :WEBKIT2 :WEBKIT2-2.38 :WEBKIT2-CORS-ALLOWLIST :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56 :GLIB-2-58 :GLIB :NYXT-3.1 :NYXT-3 :NYXT-3-PRE-RELEASE-2 :NYXT-3-PRE-RELEASE-2-191-GD6BC6D9DC :CUSTOM-HASH-TABLE-NATIVE :SLYNK :PLUMP-UTF-32 :PARENSCRIPT :FSET-EXT-STRINGS :GLOBAL-VARS :DECLARE-TYPES :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :SWANK :LPARALLEL :21BIT-CHARS :CLOSER-MOP :CL-PPCRE-UNICODE :CL-UNICODE :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED :CL-PPCRE :CHUNGA :FLEXI-STREAMS :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO :CL-JSON-CLOS :CL-JSON :SBCL-USES-SB-ROTATE-BYTE CHIPZ-SYSTEM:GRAY-STREAMS :THREAD-SUPPORT :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.1
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY)
Critical dependencies: (/home/build/nyxt-browser-git/src/nyxt-browser/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd /home/build/nyxt-browser-git/src/nyxt-browser/_build/cl-gobject-introspection/cl-gobject-introspection.asd /home/build/nyxt-browser-git/src/nyxt-browser/_build/cl-webkit/webkit2/cl-webkit2.asd)

```

Cheers and keep it up!

**Comment by aartaka** [comment]
@OlMon, yes, I noticed hint disappearance too!

@aadcg, it might be a regression after #2192.

**Comment by aadcg** [comment]
That's not the problem. 

The problem goes back to commit 2dbf2ab9f, where @Ambrevar shortsightedly removed duplicates from featuring at the prompt buffer but disregarded the fact that they're still being drawn in the UI.

**Comment by aartaka** [comment]
Ooops. So do we revert it then?

**Comment by aartaka** [comment]
And, actually, there may be several buttons with same URLs but different side-effects, so it's not cool to remove duplicates from hints.

I should've told it earlier, but I only realised it now :)

**Comment by aadcg** [comment]
My opinion is that de-duplication is certainly useful and desirable, but as of today we lack some prerequisites to make a clean implementation.

Let me revert it myself, as it's not a simple matter of reverting one commit. After the aforementioned commit, I've also worked on top of it since the de-duplication logic on the prompt buffer was rather slow.

Also, I can (re-)open the issue on the de-duplication topic and warn against some pitfalls.

**Comment by aartaka** [comment]
Yes, thank you `(‘∀’●)♡`

**Comment by aadcg** [comment]
To be clear, I've only put some more thought into this change now that this issue was opened, and haven't noticed it before!

I think the take-away is that the slighting innocent change might have "spooky actions at a distance", and I say this against myself! To mitigate this, we certainly need more testing and opening PRs as the general rule.

**Comment by aadcg** [comment]
Closed via da6853b65.
